qiskit/circuit/library/n_local/n_local.py

```diff
@@ -78,134 +78,170 @@ def n_local(
+      qubits are swapped every block (therefore alternating).
+
+    If an entanglement layer contains multiple blocks, then the entanglement should be
+    given as list of entanglements for each block. For example::
+
+        entanglement_blocks = ["rxx", "ryy"]
+        entanglement = ["full", "linear"]  # full for rxx and linear for ryy
+
+    or::
+
+        structure_rxx = [[0, 1], [2, 3]]
+        structure_ryy = [[0, 2]]
+        entanglement = [structure_rxx, structure_ryy]
+
+    Finally, the entanglement can vary in each repetition of the circuit. For this, we
+    support passing a callable that takes as input the layer index and returns the entanglement
+    for the layer in the above format. See the examples below for a concrete example.
+
+    The rotation and entanglement gates can be specified via single strings, if they
+    are made up of a single block per layer:
+
+    .. plot::
+        :alt: Circuit diagram output by the previous code.
+        :include-source:
+
+        from qiskit.circuit.library import n_local
+        circuit = n_local(3, "ry", "cx", "linear", reps=2, insert_barriers=True)
+        circuit.draw("mpl")
+
+    Multiple gates per layer can be set by passing a list. Here, for example, we use
+    Pauli-Y and Pauli-Z rotations in the rotation layer:
+
+    .. plot::
+        :alt: Circuit diagram output by the previous code.
+        :include-source:
+
+        from qiskit.circuit.library import n_local
+        circuit = n_local(3, ["ry", "rz"], "cz", "full", reps=1, insert_barriers=True)
+        circuit.draw("mpl")
+
+    To omit rotation or entanglement layers, the block can be set to an empty list:
+
+    .. plot::
+        :alt: Circuit diagram output by the previous code.
+        :include-source:
+
+        from qiskit.circuit.library import n_local
+        circuit = n_local(4, [], "cry", reps=2)
+        circuit.draw("mpl")
+
+    The entanglement can be set explicitly via the ``entanglement`` argument:
+
+    .. plot::
+        :alt: Circuit diagram output by the previous code.
+        :include-source:
+
+        from qiskit.circuit.library import n_local
+        entangler_map = [[0, 1], [2, 0]]
+        circuit = n_local(3, "x", "crx", entangler_map, reps=2)
+        circuit.draw("mpl")
+
+    We can set different entanglements per layer, by specifing a callable that takes
+    as input the current layer index, and returns the entanglement structure. For example,
+    the following uses different entanglements for odd and even layers:
+
+    .. plot::
+        :alt: Circuit diagram output by the previous code.
+        :include-source:
+
+        from qiskit.circuit.library import n_local
+        def entanglement(layer_index):
+            if layer_index % 2 == 0:
+                return [[0, 1], [0, 2]]
+            return [[1, 2]]
+        circuit = n_local(3, "x", "cx", entanglement, reps=3, insert_barriers=True)
+        circuit.draw("mpl")
+
+    Additional usage examples::
```

Comment:
You just dumped the examples in addition here, but most of them are already covered. Duplicate documentation can make information hard to find and is not very useful. 
* Please reduce this to only examples that do not exist yet and add value, I believe the only one not covered is about multiple rotation/entanglement blocks.
* Please ensure to match the existing documentation style, instead of just adding this as "additional examples" without explanations.